libflux/src/flux/semantic/builtins.rs

```diff
@@ -275,6 +275,267 @@ pub fn builtins() -> Builtins<'static> {
+                "#),
+                "map" => Node::Builtin("forall [t0, t1] (<-tables: [t0], fn: (r: t0) -> t1, ?mergeKey: bool) -> [t1]"),
+                "max" => Node::Builtin("forall [t0] where t0: Row (<-tables: [t0], ?column: string) -> [t0]"),
+                "mean" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?column: string
+                    ) -> [t1]
+                "#),
+                "min" => Node::Builtin("forall [t0] where t0: Row (<-tables: [t0], ?column: string) -> [t0]"),
+                "mode" => Node::Builtin(r#"
+                    forall [t0, t1, t2] where t0: Row, t2: Row (
+                        <-tables: [t0],
+                        ?column: string
+                    ) -> [{_value: t1 | t2}]
+                "#),
+                "movingAverage" => Node::Builtin("forall [t0, t1] where t0: Numeric (<-tables: [{_value: t0 | t1}], n: int) -> [{_value: float | t1}]"),
+                "pivot" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        rowKey: [string],
+                        columnKey: [string],
+                        valueColumn: string
+                    ) -> [t1]
+                "#),
+                "quantile" => Node::Builtin(r#"
+                    forall [t0] where t0: Row (
+                        <-tables: [t0],
+                        ?column: string,
+                        q: float,
+                        ?compression: float,
+                        ?method: string
+                    ) -> [t0]
+                "#),
+                // start and stop should be able to constrained to time or duration with a kind constraint:
+                // https://github.com/influxdata/flux/issues/2243
+                "range" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t3: Row (
+                        <-tables: [t0],
+                        start: t1,
+                        ?stop: t2,
+                        ?timeColumn: string,
+                        ?startColumn: string,
+                        ?stopColumn: string
+                    ) -> [t3]
+                "#),
+                "reduce" => Node::Builtin(r#"
+                    forall [t0, t1, t2] where t0: Row, t1: Row, t2: Row (
+                        <-tables: [t0],
+                        fn: (r: t0, accumulator: t1) -> t1,
+                        identity: t1
+                    ) -> [t2]
+                "#),
+                "relativeStrengthIndex" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        n: int,
+                        ?columns: [string]
+                    ) -> [t1]
+                "#),
+                // Either fn or columns should be specified
```

Comment:
Is it worth trying to find a way to annotate this?

Reply:
There are several functions that have this problem, and I think probably we'll need to change the functions in some way to make them friendlier to our type system, rather than increasing its complexity.

Josh filed a similar issue here: https://github.com/influxdata/flux/issues/1660

I think I'll file an umbrella issue for the "mutually exclusive parameters" problem, and document all the examples I can find.

Reply:
Thank you.

Reply:
There weren't too many so I just filed separate issues for each instance.  Probably they should each be handled individually.